librz/hash/state.c

```diff
@@ -47,12 +47,194 @@ RZ_API void rz_hash_do_begin(RzHash *ctx, ut64 flags) {
 	ctx->rst = false;
 }
 
+#define HANDLE_CRC_PRESET(rbits, aname) \
+	do { \
+		if (algobit & RZ_HASH_##aname) { \
+			ut##rbits res = rz_hash_crc_preset(buf, len, CRC_PRESET_##aname); \
+			rz_write_be##rbits(ctx->digest, res); \
+		} \
+	} while (0)
+
+RZ_API void rz_hash_do_update(RzHash *ctx, ut64 algobit, const ut8 *buf, ut64 len) {
```

Comment:
Not sure how it's done in the rest of the hash codebase, but this function does not look right.

If you have a plugin-based system, calling plugin-specific functions directly seems the wrong thing to do, because that essentially makes the plugins-based system useless (indeed, you can't *plug* anything unless you recompile all the code.

Reply:
i agree. this way of handling crypto is stupid.
i could either rewrite the way the whole crypto plugin thing works now or do it in another PR.

Reply:
resolved on MM